packages/cojson/src/coValueState.ts

```diff
@@ -0,0 +1,107 @@
+import { CoValueCore } from "./coValueCore.js";
+import { PeerID } from "./sync.js";
+
+function createResolvablePromise<T>() {
+    let resolve!: (value: T) => void;
+
+    const promise = new Promise<T>((res) => {
+        resolve = res;
+    });
+
+    return { promise, resolve };
+}
+
+class CoValueUnknownState {
+    type = "unknown" as const;
+    private peers: Map<PeerID, ReturnType<typeof createResolvablePromise<"available" | "unavailable">>>;
+    private resolve: (value: "available" | "unavailable") => void;
+
+    ready: Promise<"available" | "unavailable">;
```

Comment:
Renamed the states:
 'loading' -> 'unknown'
 'loaded' -> 'available'
 
Reasoning about the names helped me in the understanding process.
Happy to revert the change if the previous ones have a better fit in this context.

Reply:
This is better, what was perviously "loaded" isn't actually loaded, it's just "we've received the first content for it" - so "available" is much better

In the future we should have some kind of "loaded" state meaning "up-to-date as per all peer states"